exonum-java-binding-core/src/main/java/com/exonum/binding/service/Service.java

```diff
@@ -118,4 +118,13 @@
    * @param router a router responsible for handling requests to this service
    */
   void createPublicApiHandlers(Node node, Router router);
+
+  /**
+   * Handles block commit event. This handler is a callback method which is invoked by
+   * the blockchain after each block commit.
+   * For example, a service can create one or more transactions if a specific condition
+   * has occurred.
+   * @param event the current node state on which the blockchain is running
+   */
+  default void afterCommit(BlockCommittedEvent event) {}
```

Comment:
Is default without implementation to allow optional afterCommit()  implementation by the service ?

Reply:
Yes! Some services might not need this, so I thought default method would be better rather than forcing the user to override it with empty implementation